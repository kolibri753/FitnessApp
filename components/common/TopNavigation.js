import React from "react";
import { View, Text, StyleSheet, TouchableOpacity } from "react-native";
import { AntDesign } from "@expo/vector-icons";
import { colors } from "../../styles/colors";

const TopNavigation = ({ title, activeDot, navigation }) => {
	return (
		<View style={styles.header}>
			{navigation && (
				<TouchableOpacity onPress={() => navigation.goBack()}>
					<AntDesign name="arrowleft" size={24} color="white" />
				</TouchableOpacity>
			)}
			<Text style={styles.headerText}>{title}</Text>
			<View style={styles.steps}>
				<View style={[styles.dot, activeDot === 1 && styles.activeDot]}></View>
				<View style={[styles.dot, activeDot === 2 && styles.activeDot]}></View>
			</View>
		</View>
	);
};

const styles = StyleSheet.create({
	header: {
		height: 50,
		backgroundColor: colors.black,
		alignItems: "center",
		justifyContent: "space-between",
		paddingHorizontal: 20,
		flexDirection: "row",
		borderBottomWidth: 1,
		borderBottomColor: "white",
		borderBottomStyle: "solid",
		zIndex: 10,
	},
	headerText: {
		fontSize: 24,
		color: colors.white,
	},
	steps: {
		flexDirection: "row",
	},
	dot: {
		width: 10,
		height: 10,
		borderRadius: 5,
		marginHorizontal: 5,
		backgroundColor: colors.white,
	},
	activeDot: {
		backgroundColor: colors.yellow,
	},
});

export default TopNavigation;
