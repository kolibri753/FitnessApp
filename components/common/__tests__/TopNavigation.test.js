import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import renderer from "react-test-renderer";
import TopNavigation from '../TopNavigation';

describe('TopNavigation', () => {
  it('renders correctly with title', () => {
    const { getByText } = render(<TopNavigation title="Test Title" />);
    expect(getByText('Test Title')).toBeTruthy();
  });

  it('renders back button and calls navigation.goBack when navigation prop is provided', () => {
    const mockNavigation = { goBack: jest.fn() };
    const { getByTestId } = render(
      <TopNavigation title="Test Title" navigation={mockNavigation} />
    );
    const backButton = getByTestId('backButton');
    fireEvent.press(backButton);
    expect(mockNavigation.goBack).toHaveBeenCalled();
  });

  it("renders correctly", () => {
		const tree = renderer.create(<TopNavigation />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
