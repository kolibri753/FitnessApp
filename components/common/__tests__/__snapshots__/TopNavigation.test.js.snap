// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNavigation renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "borderBottomColor": "white",
      "borderBottomStyle": "solid",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "space-between",
      "paddingHorizontal": 20,
      "zIndex": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontSize": 24,
      }
    }
  />
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 5,
            "height": 10,
            "marginHorizontal": 5,
            "width": 10,
          },
          false,
        ]
      }
    />
    <View
      style={
        [
          {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 5,
            "height": 10,
            "marginHorizontal": 5,
            "width": 10,
          },
          false,
        ]
      }
    />
  </View>
</View>
`;
